src/controllers/authController.ts

```diff
@@ -0,0 +1,189 @@
+import { Request, Response } from 'express';
+import bcrypt from 'bcrypt';
+import jwt, { JwtPayload, VerifyErrors, VerifyOptions } from 'jsonwebtoken';
+import UsersDB from '../models/user';
+import mongoose from 'mongoose';
+
+// Custom Type
+import User from '../types/userType';
+
+// Custom Runtypes
+import { RequestBodyForLogin, RequestBodyWithDOB, RequestBodyWithoutDOB } from '../types/runtypes';
+
+// Sign-Up a new user
+const signup = async(req: Request<{}, {}, User>, res: Response) => {

```

Comment:
if possible add return type declaration for functions ?

Reply:
Done